buildSrc/src/main/groovy/airbyte-python-docker.gradle

```diff
@@ -0,0 +1,299 @@
+            scope = 'VIRTUALENV'
+            installVirtualenv = true
+            pip 'pip:21.3.1'
+            pip 'mccabe:0.6.1'
+            // https://github.com/csachs/pyproject-flake8/issues/13
+            pip 'flake8:4.0.1'
+            // flake8 doesn't support pyproject.toml files
+            // and thus there is the wrapper "pyproject-flake8" for this
+            pip 'pyproject-flake8:0.0.1a2'
+            pip 'black:22.3.0'
+            pip 'mypy:0.930'
+            pip 'isort:5.6.4'
+            pip 'pytest:6.1.2'
+            pip 'coverage[toml]:6.3.1'
+        }
+
+
+        project.task('isortFormat', type: PythonTask) {
+            module = "isort"
+            command = "--settings-file=${project.rootProject.file('pyproject.toml').absolutePath} ./"
+        }
+
+        project.task('isortReport', type: PythonTask) {
+            module = "isort"
+            command = "--settings-file=${project.rootProject.file('pyproject.toml').absolutePath} --diff --quiet ./"
+            outputPrefix = ''
+        }
+
+        project.task('blackFormat', type: PythonTask) {
+            module = "black"
+            // the line length should match .isort.cfg
+            command = "--config ${project.rootProject.file('pyproject.toml').absolutePath} ./"
+            dependsOn project.rootProject.licenseFormat
+            dependsOn project.isortFormat
+        }
+
+        project.task('blackReport', type: PythonTask) {
+            module = "black"
+            command = "--config ${project.rootProject.file('pyproject.toml').absolutePath} --diff --quiet ./"
+            outputPrefix = ''
+        }
+
+        project.task('flakeCheck', type: PythonTask, dependsOn: project.blackFormat) {
+            module = "pflake8"
+            command = "--config ${project.rootProject.file('pyproject.toml').absolutePath} ./"
+        }
+
+        project.task('flakeReport', type: PythonTask) {
+            module = "pflake8"
+            command = "--exit-zero --config ${project.rootProject.file('pyproject.toml').absolutePath} ./"
+            outputPrefix = ''
+        }
+
+        project.task("mypyReport", type: Exec) {
+            commandLine = ".venv/bin/python"
+            args "-m", "mypy", "--config-file", "${project.rootProject.file('pyproject.toml').absolutePath}", "./"
+            setIgnoreExitValue true
+        }
+
+
+        // attempt to install anything in requirements.txt. by convention this should only be dependencies whose source is located in the project.
+
+        if (project.file('requirements.txt').exists()) {
+            throw new GradleException("Unexpected")
+        } else if (project.file('setup.py').exists()) {
+            project.task("installLocalReqs", type: Exec) {
+                // FIXME: use pwd to get the path
+                commandLine "docker", "run", "-v", "/Users/alex/code/airbyte/airbyte-connector-builder-server:/tmp", "--entrypoint", "pip", "python:3.9-slim", "install", "-e", "/tmp"
```

Comment:
@sherifnada this is how we can run the python command from inside a docker container